### DISCUSSION TITLE
I want to know what technology is used to implement the underlying technology of Peer Systems. I did not find the corresponding code in the source code.

### DISCUSSION THREAD

**Discussion by robinBeanMaker** [Keyfactor/ejbca-ce#557] [Q&A]
I want to know what technology is used to implement the underlying technology of Peer Systems. I did not find the corresponding code in the source code.

**Comment by primetomas** [comment]
Peer connectors is an Enterprise feature, so you will not find the source here.
It's documented here: https://doc.primekey.com/ejbca/ejbca-operations/ejbca-operations-guide/ca-operations-guide/peer-systems-operations

**Reply by robinBeanMaker** [reply]
Do Peer Systems communicate two-way?
For example, if CA registers with RA through Peer Systems, how does RA synchronize information to CA? Because RA, as a server, does not have the ability to push messages to CA.

**Reply by primetomas** [reply]
It is fairly detailed described here: https://doc.primekey.com/ejbca/ejbca-operations/ejbca-ca-concept-guide/peer-systems

Is there any information you miss on that page?
